smithy-jmespath/src/main/java/software/amazon/smithy/jmespath/ast/IndexExpression.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.smithy.jmespath.ast;
+
+import java.util.Objects;
+import software.amazon.smithy.jmespath.ExpressionVisitor;
+import software.amazon.smithy.jmespath.JmespathExpression;
+
+/**
+ * Gets a specific element by zero-based index. Use -1 to get the
```

Comment:
This should explain `-` as used for from end, instead of just `-1` being possible.